$OpenBSD: patch-src_decoder_OggUtil_cxx,v 1.1 2014/05/06 10:49:23 dcoppa Exp $

commit 70bd35abe2ab774b70e37822f0200fcd4ebcd6c6
Author: Max Kellermann <max@duempel.org>
Date:   Tue Apr 29 11:56:05 2014 +0200

decoder/OggUtil: allow skipping up to 32 kB after seek

Fixes missing song length on high-latency Opus files.

According to tests with 320 kbit/s opus files with 60ms packets, we
need to skip up to 29 kB.

--- src/decoder/OggUtil.cxx.orig	Tue May  6 12:27:40 2014
+++ src/decoder/OggUtil.cxx	Tue May  6 12:28:15 2014
@@ -81,7 +81,7 @@ bool
 OggExpectPageSeek(ogg_sync_state &oy, ogg_page &page,
 		  Decoder *decoder, InputStream &input_stream)
 {
-	size_t remaining_skipped = 16384;
+	size_t remaining_skipped = 32768;
 
 	while (true) {
 		int r = ogg_sync_pageseek(&oy, &page);
